File: scripts/download-data.mjs
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";
import { defaultConfig, fileStem, parseArgs } from "../backtest/config.mjs";
import { downloadOkxHistory } from "../backtest/okx.mjs";
import { writeJson } from "../backtest/io.mjs";

const root = dirname(dirname(fileURLToPath(import.meta.url)));
const config = parseArgs(process.argv.slice(2), defaultConfig);
const outputPath = join(root, "data", "raw", `${fileStem(config)}_raw.json`);

const rawPayload = await downloadOkxHistory(config);
await writeJson(outputPath, rawPayload);

console.log(JSON.stringify({
  step: "download",
  outputPath,
  instrument: rawPayload.instrument,
  bar: rawPayload.bar,
  rowCount: rawPayload.rowCount,
  pageCount: rawPayload.pageCount,
  downloadedAt: rawPayload.downloadedAt
}, null, 2));

File: backend_py/download_data.py
from __future__ import annotations

import json
import sys

from backend_py.data_io import write_json
from backend_py.reports_reader import DATA_RAW_DIR
from backend_py.research.config import file_stem, parse_args
from backend_py.research.okx import download_okx_history


def main(argv: list[str] | None = None) -> None:
    config = parse_args(list(argv if argv is not None else sys.argv[1:]))
    output_path = DATA_RAW_DIR / f"{file_stem(config)}_raw.json"
    raw_payload = download_okx_history(config)
    write_json(output_path, raw_payload)
    print(
        json.dumps(
            {
                "step": "download-py",
                "outputPath": str(output_path),
                "instrument": raw_payload["instrument"],
                "bar": raw_payload["bar"],
                "rowCount": raw_payload["rowCount"],
                "pageCount": raw_payload["pageCount"],
                "downloadedAt": raw_payload["downloadedAt"],
            },
            ensure_ascii=False,
            indent=2,
        )
    )


if __name__ == "__main__":
    main()
